### PR TITLE
gh-109286  Updated macOS and Windows installers to SQLite 3.43.1 

### PR DESCRIPTION
**Pull request by ghost**
Updated macOS and Windows installers to SQLite 3.43.1
```
I made changes to 4 files following the templated structure for this kind of update which was outlined in previous issues. The files that were changed are
Mac/BuildScript/build-installer.py
PCbuild/get_externals.bat
PCbuild/python.props
PCbuild/readme.txt
```

The issue is  gh-109286

### CODE REVIEW

**Comment by jtranquilli** [comment]
Seems to be having an issue with the checksum variable. I copied that hash code directly from the site : https://sqlite.org/download.html. It is the correct hash assigned to the .tar.gz file for the source code download for version 3.43.1.

File: Mac/BuildScript/build-installer.py [hunk 1]
```diff
@@ -359,10 +359,9 @@ def library_recipes():
                   ),
           ),
           dict(
-              name="SQLite 3.42.0",
-              url="https://sqlite.org/2023/sqlite-autoconf-3420000.tar.gz",
-              checksum="0c5a92bc51cf07cae45b4a1e94653dea",
-              extra_cflags=('-Os '
+              name="SQLite 3.43.1",
+              checksum="fd32866c281539eae95cd90b5c2c34337d8808822a988211b9b009c1e788e42d",      
+	      extra_cflags=('-Os '
```
Comment (felixxm) [inline Mac/BuildScript/build-installer.py]: Please revert this unrelated indentation change.

File: Mac/BuildScript/build-installer.py [hunk 2]
```diff
@@ -359,10 +359,9 @@ def library_recipes():
                   ),
           ),
           dict(
-              name="SQLite 3.42.0",
-              url="https://sqlite.org/2023/sqlite-autoconf-3420000.tar.gz",
-              checksum="0c5a92bc51cf07cae45b4a1e94653dea",
-              extra_cflags=('-Os '
+              name="SQLite 3.43.1",
+              checksum="fd32866c281539eae95cd90b5c2c34337d8808822a988211b9b009c1e788e42d",
```
Comment (felixxm) [inline Mac/BuildScript/build-installer.py]: Is there a reason why you removed `url`?

Comment (felixxm) [inline Mac/BuildScript/build-installer.py]: This should be a md5 checksum of `https://sqlite.org/2023/sqlite-autoconf-3430100.tar.gz`.

**Review by felixxm — COMMENTED** [review]
@jtranquilli Thanks :+1: Please split it into two PRs, one for Windows and one for macOS installer.

**Comment by zware** [comment]
> Please split it into two PRs, one for Windows and one for macOS installer.

Doing both in one PR is fine.

There are several failing checks right now, though:

- CLA Signing — Please sign our Contributor License Agreement.
Required
[Details](https://cpython-clabot.herokuapp.com/contributor-license-agreement?state=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJnaXRIdWJVc2VySWQiOjc2MjMxMTIwLCJhZ3JlZW1lbnRWZXJzaW9uSWQiOm51bGwsImF1dGhvcnMiOlsianVsaXVzdHJhbnF1aWxsaUBKdWxpdXNzLU1hY0Jvb2stUHJvLmxvY2FsIl0sInB1bGxSZXF1ZXN0Ijp7ImlkIjoxNTM5MTYzNDYxLCJudW1iZXIiOjExMDI1MiwidXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL3B5dGhvbi9jcHl0aG9uL3B1bGwvMTEwMjUyIiwiaGVhZFNoYSI6ImMzMjBlZDJmZDhkZTc1Yzc5YzBkNmQwYTI3ZmQ4NWRhNDU0MzVlYjQifSwicmVwb3NpdG9yeSI6eyJpZCI6ODE1OTg5NjEsIm93bmVyIjoicHl0aG9uIiwib3duZXJJZCI6MTUyNTk4MSwibmFtZSI6ImNweXRob24iLCJmdWxsTmFtZSI6InB5dGhvbi9jcHl0aG9uIn0sImlhdCI6MTY5NjMwMjU1N30.QzJ07ZDfjC7vADANCrgZJMS7QmXFvcr-eXUxdIaOmK4) 
  - This is the most important one; we cannot accept your contribution without a signed license agreement.  You may need to configure `user.email` correctly in your local `.gitconfig` file and recreate your commits with the correct user information.
- bedevere/issue-number — No issue # in title or "skip issue" label found
Required
[Details](https://devguide.python.org/getting-started/pull-request-lifecycle.html#submitting)
  - This one just needs a rearrangement of the PR title, it should be of the form `gh-###: <short description>`
- Lint / lint (pull_request) Failing after 36s
[Details](https://github.com/python/cpython/actions/runs/6388173722/job/17337596021?pr=110252)
  - You can run either `make patchcheck` or `pre-commit` locally, or just copy and apply the patch provided in the output at the `Details` link.

**Comment by zooba** [comment]
We typically have two separate NEWS entries for these changes, as they affect the release more than build. You can add two entries in the same PR (just change the random characters near the end of the filename), but people are probably going to be more comfortable with one NEWS entry per PR, just because they won't have seen multiple ones so often.

File: Mac/BuildScript/build-installer.py (hunk 1 above)
Comment (ghost) [reply Mac/BuildScript/build-installer.py]: Now fixed on my end, will push soon.

File: Mac/BuildScript/build-installer.py (hunk 2 above)
Comment (ghost) [reply Mac/BuildScript/build-installer.py]: Sorry about that, fixed on my end now, will push after I address the other issues here.

Comment (ghost) [reply Mac/BuildScript/build-installer.py]: Fixed on my end, will be pushing soon.

**Comment by erlend-aasland** [comment]
I usually split these PRs, since the response time of the Windows and macOS teams can vary greatly :)

File: Misc/NEWS.d/next/Build/2023-10-03-03-09-11.gh-issue-109268.HFLSeh.rst
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Build/2023-10-03-03-09-11.gh-issue-109268.HFLSeh.rst:1]: Please remove this and add two NEWS items; one for macOS, and one for Windows. Also note that it is _macOS_, not _MacOS_.

**Review by erlend-aasland — CHANGES_REQUESTED** [review]

**Comment by jtranquilli** [comment]
> > Please split it into two PRs, one for Windows and one for macOS installer.
> 
> Doing both in one PR is fine.
> 
> There are several failing checks right now, though:
> 
> * CLA Signing — Please sign our Contributor License Agreement.
>   Required
>   [Details](https://cpython-clabot.herokuapp.com/contributor-license-agreement?state=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJnaXRIdWJVc2VySWQiOjc2MjMxMTIwLCJhZ3JlZW1lbnRWZXJzaW9uSWQiOm51bGwsImF1dGhvcnMiOlsianVsaXVzdHJhbnF1aWxsaUBKdWxpdXNzLU1hY0Jvb2stUHJvLmxvY2FsIl0sInB1bGxSZXF1ZXN0Ijp7ImlkIjoxNTM5MTYzNDYxLCJudW1iZXIiOjExMDI1MiwidXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL3B5dGhvbi9jcHl0aG9uL3B1bGwvMTEwMjUyIiwiaGVhZFNoYSI6ImMzMjBlZDJmZDhkZTc1Yzc5YzBkNmQwYTI3ZmQ4NWRhNDU0MzVlYjQifSwicmVwb3NpdG9yeSI6eyJpZCI6ODE1OTg5NjEsIm93bmVyIjoicHl0aG9uIiwib3duZXJJZCI6MTUyNTk4MSwibmFtZSI6ImNweXRob24iLCJmdWxsTmFtZSI6InB5dGhvbi9jcHl0aG9uIn0sImlhdCI6MTY5NjMwMjU1N30.QzJ07ZDfjC7vADANCrgZJMS7QmXFvcr-eXUxdIaOmK4)
>   
>   * This is the most important one; we cannot accept your contribution without a signed license agreement.  You may need to configure `user.email` correctly in your local `.gitconfig` file and recreate your commits with the correct user information.
> * bedevere/issue-number — No issue # in title or "skip issue" label found
>   Required
>   [Details](https://devguide.python.org/getting-started/pull-request-lifecycle.html#submitting)
>   
>   * This one just needs a rearrangement of the PR title, it should be of the form `gh-###: <short description>`
> * Lint / lint (pull_request) Failing after 36s
>   [Details](https://github.com/python/cpython/actions/runs/6388173722/job/17337596021?pr=110252)
>   
>   * You can run either `make patchcheck` or `pre-commit` locally, or just copy and apply the patch provided in the output at the `Details` link.

I just updated my .gitconfig file to use my GitHub email so that I should be able to sign the CLA now, but this PR is still associated with some other default email. Is the best way forward to create a new pull request with the updated associated email or is there some way to salvage this request?

**Comment by zware** [comment]
Depends on the level of Git wizardry that you're comfortable with :).  You can recreate the commits and force-push, or you can take the opportunity to split it up and push fresh commits to two new branches and thus new PRs.
